redwood-lazylayout-widget/src/commonMain/kotlin/app/cash/redwood/lazylayout/widget/LazyListUpdateProcessor.kt

```diff
@@ -172,6 +191,7 @@ public abstract class LazyListUpdateProcessor<V : Any, W : Any> {
         edit is Edit.Insert &&
         edit.index == loadedItems.size
       ) {
+        maybeShiftLoadedWindow(newItemsBefore + loadedItems.size)
```

Comment:
To consider for follow-up . . . shift the loaded window when converting from loaded items to placeholders, instead of the opposite.